vespamalloc/src/vespamalloc/malloc/allocchunk.h

```diff
@@ -61,16 +61,20 @@ struct AtomicTaggedPtr {
     void *_ptr;
     size_t _tag;
 } __attribute__ ((aligned (16)));
+
+using TaggedPtr = AtomicTaggedPtr;
+
 #else
-    using AtomicTaggedPtr = TaggedPtr;
+    using TaggedPtr = TaggedPtrT;
+    using AtomicTaggedPtr = std::atomic<TaggedPtr>;
 #endif
 
 
 class AFListBase
 {
 public:
-    using HeadPtr = std::conditional<std::atomic<TaggedPtr>::is_always_lock_free, TaggedPtr, AtomicTaggedPtr>::type;
-    using AtomicHeadPtr = std::conditional<std::atomic<TaggedPtr>::is_always_lock_free, std::atomic<TaggedPtr>, AtomicTaggedPtr>::type;
+    using HeadPtr = TaggedPtr;
+    using AtomicHeadPtr = std::atomic<TaggedPtr>;
```

Comment:
Did this use to be `std::atomic` on x64? I was under the impression that `is_always_lock_free` is `false` for 128-bit x64 atomics due to the whole "can't use 128-bit CaS since it doesn't work with read-only memory" debacle. If so, can this cause any performance regressions due to an introduction of implicit spinlocking?